modules/rntuple.mjs

```diff
@@ -364,18 +364,72 @@ _readClusterGroups(reader) {
     numClusters = reader.readU32();
 
     console.log(`Cluster Record Size: ${clusterRecordSize}`);
-    console.log(`Min Entry: ${minEntry}, Entry Span: ${entrySpan}, Num Clusters: ${numClusters}`);
-
-    clusterGroups.push({
+    
+    // Locator method to get the page list locator offset
+    const pageListLocator = this._readLocator(reader);
+
+  // Seek to the page list offset
+  reader.seek(pageListLocator.fOffset);
+  console.log('Page List Locator:', pageListLocator);
+  console.log(`Page List Locator Offset (hex): 0x${pageListLocator.offset.toString(16).toUpperCase()}`);
+  // Deserialize the Page List Envelope from there
+  this._readPageListEnvelope(reader);
+ const group = {
       minEntry,
       entrySpan,
       numClusters,
-    });
+      
+    }; 
+  
+    clusterGroups.push(group);
   }
-
   this.clusterGroups = clusterGroups;
 }
 
+_readLocator(reader) {
+  const sizeAndType = reader.readU32(),            // 4 bytes: size + T bit
+  type = sizeAndType & 1,                    // last bit
+  size = sizeAndType >>> 1,                 // top 31 bits
```

Comment:
This is wrong: the spec says:
> Size: If T is zero, the number of bytes to read, i.e. the compressed size of the referenced block. Otherwise, the 16 least-significant bits, i.e. bits 0:15, specify the size of the locator itself.

So `size` is actually equal to `sizeAndType` if `type === 0` (if it's not you throw the error so you don't have to handle it right now)